contrib/format-xml/src/main/java/org/apache/drill/exec/store/xml/xsd/DrillXSDSchemaVisitor.java

```diff
@@ -59,10 +77,34 @@ public TupleMetadata getDrillSchema() {
     return builder.build();
   }
 
+  /**
+   * Handles global elements establishing a map for the child elements and attributes (if any).
+   * <p/>
+   * TBD: Does not handle case where multiple elements have the same name as in:
+   * <pre>{@code
+   * <element name="a" .../>
+   * <element name="b" .../>
+   * <element name="a" .../>
+   * }</pre>
+   * There is also the case where they are ambiguous unless namespaces are used:
+   * <pre>{@code
+   * <element name="a" .../>
+   * <element ref="pre:a" .../> <!-- without namespace, ambiguous with prior "a" -->
+   * }</pre>
+   */
   @Override
   public void onEnterElement(XmlSchemaElement xmlSchemaElement, XmlSchemaTypeInfo xmlSchemaTypeInfo, boolean b) {
+    assert nestingLevel >= 0;
     boolean isRepeated = xmlSchemaElement.getMaxOccurs() > 1;
     String fieldName = xmlSchemaElement.getName();
+    //
+    // Note that the child name in constant ATTRIBUTE_MAP_NAME is reserved and cannot be used
+    // by any child element.
+    //
```

Comment:
E.g., right here: what is the right way to error out here.

Reply:
Something like this:

```java
throw UserException
        .dataReadError(e)
        .message(String.format("Failed to open input file: %s", file.split().getPath().toString()))
        .addContext(errorContext)
        .addContext(e.getMessage())
        .build(logger);
```

The `UserException` class has various errors which you can throw.   Also when used in the context of a file reader, you have another object called the `errorContext` which the file reader populates.  This actually contains a lot of useful information such as the file path, storage plugin etc.  It was all part of a greater effort to provide more helpful error messages.

Passing the `errorContext` is not strictly necessary, but helpful for the users to debug what went wrong.

Reply:
What is the right ErrorType for a Schema error? I.e, the data source uses schemas (of some sort), and there is something wrong with the schema, so it cannot be used/work, uses an unsupported feature of the schema language, or the schema is simply malformed?

Should I create one?